src/main/dynamo/QueryBuilder.ts

```diff
@@ -1,21 +1,27 @@
 import { JayZ } from "@ginger.io/jay-z"
```

Comment:
This commit extracts some functions and types into a `pagedIterator` file. The motivation here is that we already had logic to paginate DynamoDB results for `query` operations. And since the DynamoDB API for `scan` is nearly identical, we can reuse that pagination code.